javascript/modules/client/src/internal/client-addressList/decoding.ts

```diff
@@ -0,0 +1,71 @@
+import { bytesToHex } from "@aragon/sdk-common";
+import { ContextPlugin } from "../../context-plugin";
+import { ClientCore } from "../core";
+import {
+  decodeAddMemebersAction,
+  decodeRemoveMemebersAction,
+  decodeUpdatePluginSettingsAction,
+  getFunctionFragment,
+} from "../encoding/plugins";
+import { IInterfaceParams } from "../interfaces/common";
+import {
+  IClientAddressListDecoding,
+  IPluginSettings,
+} from "../interfaces/plugins";
+
+export class ClientAddressListDecoding extends ClientCore
+  implements IClientAddressListDecoding {
+  constructor(context: ContextPlugin) {
+    super(context);
+  }
+
+  /**
+   * Decodes a dao metadata from an encoded update metadata action
+   *
+   * @param {Uint8Array} data
+   * @return {*}  {IPluginSettings}
+   * @memberof ClientAddressListDecoding
+   */
+  public updatePluginSettingsAction(data: Uint8Array): IPluginSettings {
+    return decodeUpdatePluginSettingsAction(data);
```

Comment:
Is going to create duplication in the plugins because some methods are shared `encodeUpdatePluginSettingsAction` and `decodeUpdatePluginSettingsAction` specifically. Maybe It would mad more sense once we divide the packages of the `client-erc20` and `client-addresslist`

Reply:
If you think that dividing these clients into independent packages can be a good solution, now is a good time to do it :+1: 

Also, in the case where the logic is the exact same, it makes sense to factor the common code out. But if they have, for example, different plugin parameters, then I would have such logic in place.

Reply:
Yes, maybe we can edit the [APP-1026](https://aragonassociation.atlassian.net/browse/APP-1026), because this was supposed to be to move files from address list and erc20 clients to a separate folder and increase a bit the scope of the task to make independent packages and address this kind of problems